airflow/providers/openlineage/plugins/adapter.py

```diff
@@ -117,13 +117,19 @@ def build_task_instance_run_id(dag_id, task_id, execution_date, try_number):
             )
         )
 
-    def emit(self, event: RunEvent):
+    def emit(self, event: RunEvent) -> RunEvent:
+        """Emit OpenLineage event.
+
+        :param event: Event to be emitted.
+        :return: Redacted Event.
+        """
         if not self._client:
             self._client = self.get_or_create_openlineage_client()
         redacted_event: RunEvent = self._redacter.redact(event, max_depth=20)  # type: ignore[assignment]
         try:
-            with Stats.timer("ol.emit.attempts"):
-                return self._client.emit(redacted_event)
+            with Stats.timer(f"ol.emit.attempts.{event.eventType.value}.{self._client.transport.kind}"):
+                self._client.emit(redacted_event)
+            return redacted_event
```

Comment:
`self._client.emit` returns nothing, so i changed the `emit()` to return redacted_event, this way, i can pass it to the listener and emit size event with operator name. (not sure if correct approach, but easy one 😉 )

Reply:
I think it's more natural than, for example, passing gauge, but maybe pass the task type? No strong feelings about this one.

Reply:
Was thinking about passing just a gauge or a task_type, but this seems more future proof ? I don't have any specific use case in mind tho